### DISCUSSION TITLE
Koishi 4.12.0

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#1025] [General]
## Notable Changes

- HMR 相关功能从内置拆分到 @koishijs/plugin-hmr 插件中 (koishijs/koishi@f313257d41659d12decc78544ff1cb78306229ff)
- 将 feedback, recall 两个官方插件被迁移到社区中继续维护 (koishijs/koishi@89adef5e9039a2b93b11eee7f1831df6d3025c7f)
- 引入了新的内置数据表 `binding`，用于存储账号绑定信息 (koishijs/koishi@9bf5c387fb234d12117631d2e316b86f3f153c3a)
- 引入了数据库迁移技术，支持插件在升级时将已有的数据迁移到新的表中 (shigma/minato@afe0ca259c4454530ab8ca1f5311bff3609d0a40, shigma/minato@a7cbbb03127d959fd834e4cb6d6209efda20bf81, shigma/minato@752656b0d27a6e26b449ce78f9fe253ca280202b, shigma/minato@b3e03ae767fd0ab645f3805fb3d09dcb110b5f88)
- @koishijs/helpers 被废弃，相关功能迁移至 @koishijs/plugin-admin 中 (koishijs/koishi@2a2c310b90f349ffa9b9ea5566173f33926e5838, koishijs/koishi@39d56c18c5713d74a56f210f88f06faf7d33d36a, koishijs/koishi@a76ae3d6095073aa22650160929325d2ba6f2b3f)

## Features

- **auth:** 支持了在控制台解除绑定 (koishijs/console@c7e91c625ceeb89c29b46020fc11d87b2ed34dd9)
- **bind:** 支持了同平台用户绑定 (koishijs/koishi@c5f2eb77858c52940af9c3fbe8ae6a2d60442956)
- **bind:** 支持了 `-r` 选项，用于解除绑定 (koishijs/koishi@c4f6f7cb64924a33a443c180c944528490de9681)
- **core:** 现在默认情况下指令如果抛出错误会告知调用者，此行为可以通过 `handleError` 配置 (koishijs/koishi@0ebb911ba0c4a0b8c180da5cd29e40d910f0cd24)
- **locales:** 识别本地化中的 `$` 特殊标识符以支持定制化的分页 (koishijs/console#122) (koishijs/console@51f7a7f3241449f4f7fa29f5a24307c6b71e646b)

## Bug Fixes

- **client:** 修复了 `<k-slot>` 数据丢失响应性的问题 (koishijs/console#120) (koishijs/console@25ebb6767585967a25693115fa426af13cc37bec)
- **commands:** 修复了部分指令无法跳转到插件详情页的问题 (koishijs/console#123) (koishijs/console@c4a7b69f27e57045b9ee16952d952712ef1c683f)
- **core:** 修复了部分指令未显示本地化条目的问题 (koishijs/console#121) (koishijs/koishi@ab76d04b763124b64d5114e48459cb309a487812, koishijs/koishi@f994b5b8805d74b14b8bffedbe1e6e09079b2007)
- **core:** 修复了使用相同的文本覆盖本地化也会报错的问题 (koishijs/koishi@184d6eaf74ade32da229c52ab0662b971f66400d)
- **help:** 修复了重载 help 插件报错的问题 (koishijs/koishi#1014) (koishijs/koishi@5ac6876062d752d1bd1078c3263906c9bf5d4e65)
- **status:** 修复了环境信息显示报错的问题 (koishijs/console@c68400209e07ce9021b88bc0b7af6c6bbe3531ae)

## Other Changes

- chore: bump versions (koishijs/koishi@7ff566087353cd4cf4a4b37407e8fe173260afe6)
- refa: drop exit command in favor of koishi-plugin-shutdown (koishijs/koishi@3ec508654dfddd33bc7486e0d7647e7ff606afd2)
- chore: add missing devDependencies (koishijs/koishi@aebc9b4000a33579cd8972beb7eafd3e5bf85da6)
- chore: fix typings (koishijs/koishi@a60db8254989ecdf7ab90009b31e4ae7312ddecd)
- chore: minor revisions (koishijs/koishi@148041aa8d5c11693b490454a72924916b49b8a9)
- chore: bump versions (shigma/minato@69f6b88d0ea3f4cae1d0153202649249dc841fb7)
- chore: bump versions (koishijs/console@4ddc2216726e3f38573efade8bd133acfedc640c)
- chore: bump versions (koishijs/koishi@1befe72bad1b98a44c88d23506d0a53471ffd995)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.12.0'>Koishi 4.12.0</a>.</em>

**Comment by neveler** [comment]
移除了 him
